server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -483,8 +483,14 @@ public Consumer create(
             // Share consumers do not need identity certificates so refuse to create them.
             identityCertCreation = false;
             validateShareConsumer(consumer, principal, keys);
+            // if there exists a share consumer between the two orgs, return it.
+            Consumer existingConsumer = consumerCurator.searchOwnerConsumers(owner,
+                null, null, null, null, null, null, null, null,
+                consumer.getRecipientOwnerKey()).uniqueResult();
```

Comment:
Let's create a ConsumerCurator method specifically for finding a share consumer. All the null parameters screams break out into a method. I'm sure we are doing the same sort of thing in other places, but let's fix this one.